app/code/Magento/Inventory/Model/Stock/SearchCriteria/CollectionProcessor/StockFilterProcessor/StockNameFilter.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\Stock\SearchCriteria\CollectionProcessor\StockFilterProcessor;
+
+use Magento\Framework\Api\Filter;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor\CustomFilterInterface;
+use Magento\Framework\Data\Collection\AbstractDb;
+
+class StockNameFilter implements CustomFilterInterface
+{
+
+    /**
+     * Apply Custom Filter to Collection
+     *
+     * @param Filter $filter
+     * @param AbstractDb $collection
+     * @return bool Whether the filter was applied
+     * @since 100.2.0
+     */
+    public function apply(Filter $filter, AbstractDb $collection)
+    {
+        $conditionType = $filter->getConditionType();
+        $value = $filter->getValue();
+
+        if ($conditionType === 'fulltext') {
+            $conditionType = 'like';
+            $value = '%' . $value . '%';
```

Comment:
Looks like we need to quote input value
Also, need to consider if the value will contain symbols like `%` or `_`

Reply:
For quoting are you saying that I should use the mysql quoting function like `$collection->getConnection()->quote()`? I can't find any other reference in the codebase of a 'like' condition being quoted. Can you better explain this change request?